### PR TITLE
Add an HowTo article about assembly levels

### PR DESCRIPTION
**Pull request by YohannDudouit**
This PR proposes an HowTo article for the assembly levels and the different backends in MFEM.

Resolve mfem/mfem#2313

### CODE REVIEW

File: src/howto/assembly_levels.md [hunk 1]
```diff
@@ -0,0 +1,55 @@
...
+| Gradient Integrator               | V           | X            | X               |
+| Vector Divergence Integrator      | V           | X            | X               |
+| Vector FE Divergence Integrator   | V           | X            | X               |
+| Curl Curl Integrator              | V           | X            | X               |
+| Div Div Integrator                | V           | X            | X               |
+
+| Features                          | native MFEM | OCCA backend | libCEED backend |
+|-----------------------------------|-------------|--------------|-----------------|
+| Tensor elements support           | V           | V            | V               |
+| Simplices support                 | X           | X            | V               |
```
Comment (tzanio) [inline src/howto/assembly_levels.md:50]:
```suggestion
| Simplices support                 | ❌            | ❌             | ✅                |
```

**Comment by tzanio** [comment]
Wow, that looks very nice... thanks for working on it @YohannDudouit !

File: src/howto/assembly_levels.md (hunk 1 above)
Comment (acfisher) [reply src/howto/assembly_levels.md]: Excellent work.  I like the detail and the overview table.

**Comment by YohannDudouit** [comment]
@tzanio @acfisher I added a table about device configurations, but this might belong to its own HowTo article. Tell me what you think. Also the libCEED image rendering in dark mode is awful.

**Review by acfisher — APPROVED** [review]
Looking good to me.  I'm going to pull this in.

**Pull request merged**